core/game_state.py

```diff
@@ -251,14 +256,39 @@ def set_counter(self, counter_id, value):
         self._counters[counter_id] = value
         event = CounterEvent(counter_id, CounterEventType.SET, value)
         
-    def reveal(self, location, cards, position, number):
+    def reveal(self, location, position=None, number=None):
         """
         Reveal a card or cards to all players.
 
         Parameters:
-            ToDo
+            location (`Location`): Location to reveal
+            position (`StackPosition`): Position to reveal from (e.g. `TOP`) 
+            number (`int`): Number of cards to reveal 
+        """
+        players = [agent.name() for agent in self.get_agents_besides_current_player()]
+        cards = None
+        card_stack = self.get_location(location)
+        if number and location.type == LocationName.DRAW_PILE:
+            self.shuffle_discard_in_if_insufficient_cards(number)
+        if position and number:
+            cards = card_stack.peek_from_position(position, number)
+
+        self.logger.log(
+            CardKnowledgeEvent(
+                players,
+                cards or list(card_stack),
+                location,
+                position,
+                number or card_stack.size(),
```

Comment:
number could be more than the number of cards revealed if there weren't enough cards